js/compiler.ts

```diff
@@ -55,17 +58,81 @@ interface CompilerLookup {
 interface Os {
   fetchModuleMetaData: typeof os.fetchModuleMetaData;
   exit: typeof os.exit;
+  noColor: typeof os.noColor;
 }
 
 /** Abstraction of the APIs required from the `typescript` module so they can
  * be easily mocked.
  */
 interface Ts {
+  convertCompilerOptionsFromJson: typeof ts.convertCompilerOptionsFromJson;
   createLanguageService: typeof ts.createLanguageService;
   formatDiagnosticsWithColorAndContext: typeof ts.formatDiagnosticsWithColorAndContext;
   formatDiagnostics: typeof ts.formatDiagnostics;
+  parseConfigFileTextToJson: typeof ts.parseConfigFileTextToJson;
 }
 
+/** Options that either do nothing in Deno, or would cause undesired behavior
+ * if modified. */
+const ignoredCompilerOptions: ReadonlyArray<string> = [
```

Comment:
It might be better to use a white list here, so any new compiler options are ignored by default.

Reply:
It is really 50/50.  Any new compiler option is just as likely to be useful as to be ignored.  Especially when it comes to all the variations of the _strict_ class of flags.  It is highly unlikely that a new flag would be introduced that did harm, it is far more likely that it simply would not impact the compilation.